src/parser.yy

```diff
@@ -169,10 +172,20 @@ void yyerror(bpftrace::Driver &driver, const char *s);
 %type <ast::ConfigStatementList> config_assign_stmt_list config_block
 %type <SizedType> type int_type pointer_type struct_type
 %type <ast::Variable *> var
-%type <ast::Identifier *> raw_ident
 %type <ast::Program *> program
 
 
+// A pseudo token, which is the lowest precedence among all tokens.
+//
+// This helps us explicitly lower the precedence of a given rule to force shift
+// vs. reduce, and make the grammar explicit (still ambiguous, but explicitly
+// ambiguous). For example, consider the inherently ambiguous `@foo[..]`, which
+// could be interpreted as accessing the `@foo` non-scalar map, or indexing
+// into the value of the `@foo` scalar map, e.g. `(@foo)[...]`. We lower the
```

Comment:
I'm still a bit confused. Is this clarification correct?
```suggestion
// could be interpreted as accessing the `@foo` non-scalar map (updating or reading the value at that key), or indexing
// into the value of the `@foo` scalar map, e.g. `(@foo)[...]` if it contained an array or pointer. We lower the
```

Reply:
Yes. I think whether the map contains an array or pointer is actually irrelevant, the syntax here is evaluated prior to any kind of type semantics. This is valid but ambiguous:
```
{ @val = curtask; print(@val[0]); }'
```

Presently it is the definition that is distinct from this:
```
{ @val = curtask; print((@val)[0]); }'
```